setup.py

```diff
@@ -15,6 +15,25 @@
 from pathlib import Path
 from setuptools import setup, find_packages
 
+import os
+if os.name == 'nt':
```

Comment:
Do you want this check to be run for `pip install qiskit-metal` or only for `pip install -e .`? Currently, release.yml builds and publishes a universal wheel. By default, pip will favor installing this wheel over installing the source distribution and installing from the wheel will not execute setup.py. https://github.com/Qiskit/qiskit-metal/blob/325d6114863d74edece207b2ff74b32892e9bdda/.github/workflows/release.yml#L18-L22

Reply:
ouch, alright, I want this to run with `pip install qiskit-metal` (Installs from pip PyPI), otherwise this change will be pointless. Do you know if there is a way to make the error show up in the wheel @wshanks ?

Reply:
One of the main features of wheels is that they don't execute arbitrary code. The options that I see are:

1. Remove `python setup.py bdist_wheel` from `release.yml` so that only an sdist package is published (or build Linux and macOS specific wheels but not Windows).
2. Make GeoPandas an optional dependency on Windows so `pip install qiskit-metal` does not install it and a wheel can be used on all platforms. This pushes the problem to run time. The recommended installation workflow is similar in this case though -- on Windows you need to install some extra things besides `pip install qiskit-metal` before you can use Metal. A check for GeoPandas/fiona/gdal could be put at the beginning of `qiskit_metal/__init__.py` so that lower levels of the code do not need to handle GeoPandas not being installed.

Reply:
I'm partial to option 2.

Reply:
Thank you Will, I see how option 2 is neat enough `geopandas; platform_system != "Windows"` + moving the code I have put in setup.py in the  `qiskit_metal/__init__.py`.

And I see that Jeremy's partiality might be motivated by security concern.

If these are the two options, I agree with the options 2 as well (thank you @wshanks for describing this a second time). I shall proceed with the change after testing on my machine.

Reply:
Done.
Please verify